### ISSUE TITLE
Config file discovery not working in preprocessor or CLI mode

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#148]
### Description  
The config file (`.mdbook-lint.toml`) is not being automatically discovered in either preprocessor or CLI mode. It only works when explicitly specified with `--config` flag.

#### Preprocessor Mode
1. Create a `.mdbook-lint.toml` in the book directory (where `book.toml` is)
2. Add configuration like `disabled-rules = ["MDBOOK010"]`
3. Configure preprocessor in `book.toml`:
```toml
[preprocessor.lint]  
command = "mdbook-lint"
```
4. Run `mdbook build`
5. Config is NOT picked up - rules are still applied

#### CLI Mode
1. Place `.mdbook-lint.toml` in various locations:
   - Current working directory
   - Directory containing files being linted
   - Parent directories
2. Run `mdbook-lint lint src/` 
3. Config is NOT picked up unless using `--config` flag

#### Preprocessor Mode
Should automatically discover config in this order:
1. Book directory (where `book.toml` is)
2. Parent directories up to git root
3. Support config path in `book.toml`:
```toml
[preprocessor.lint]
command = "mdbook-lint"
config = ".mdbook-lint.toml"  # relative to book root
```

#### CLI Mode
Should follow predictable search order:
1. Current working directory
2. Directory containing files being linted  
3. Parent directories up to git/project root

Similar to ESLint, Prettier, and other tools.

### Actual Behavior
- Config is completely ignored unless explicitly specified with `--config` flag
- No indication that config search is happening or failing

### Impact
- Users assume their config is being used but it's silently ignored
- Makes CI/CD setup more complex
- Inconsistent with user expectations from similar tools

### Environment
- mdbook-lint version: 0.11.1
- Operating System: macOS

### Workaround
Always use explicit `--config` flag:
```bash
mdbook-lint lint src --config .mdbook-lint.toml
```

### Related
- Need debug output to understand what's happening
- Documentation needs to clearly explain config discovery (#143)
